### PR TITLE
Configure Travis

### PR DESCRIPTION
**Pull request by advayDev1**
Based on a functional implementation I did here:
https://github.com/madvay/android-analysis-tools/commit/ea266f134bce9fdb5dec01a00f785b91ee469d09

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1 +1,29 @@
...
+
+# We can be run in a container for improved performance.
+sudo: false
+
+# If these steps fail, the build is 'errored' - i.e. misconfigured.
+# This can fail if we cannot download Gradle 2.4 or the libraries
+# we depend on.
+install: 
+  - ./gradlew wrapper
+  - ./gradlew dependencies
```
Comment (brunobowden) [inline .travis.yml]: What's the meaning of this? Is this what is run before the build and distinguishes separate errors?

File: .travis.yml [hunk 2]
```diff
@@ -1 +1,29 @@
...
+
+# If these steps fail, the build is 'failed' - i.e. we have a code defect.
+# We compile (assemble) and then build (which also tests) to capture build
+# and test failures separately.
```
Comment (brunobowden) [inline .travis.yml]: Smart idea.

**Comment by brunobowden** [comment]
LGTM. Thanks @advayDev1, this is fantastic.

It also automatically runs builds on pending PRs as well?

Fixes #178 - Travis continuous build

File: .travis.yml (hunk 1 above)
Comment (advayDev1) [inline .travis.yml]: 2 separate features of Travis:
1.  install is run first (both of the install commands, one after the other), and non-0 exits in that phase are considered 'errors' related to the environment.  if `install` fails, `script` never runs.
2.  below, build and test are run separately so we can tell manually what happened.  build vs. test don't show up differently in travis though.

install and script are poor names, it should have been 'prepare' and 'build' but that's what they gave us

**Comment by advayDev1** [comment]
yep, travis runs on:
- Every direct commit to the repo 
- Every creation of a PR
- Every update (either a replacive amended commit, or an additional commit) of every PR
- Every merge of a PR into the repo

on every branch with a .travis.yml file.

See:
as a committer to the repo:
screen shot 2015-08-20 at 1 42 34 am

as everyone else:
screen shot 2015-08-20 at 1 43 26 am

screen shot 2015-08-20 at 1 43 39 am


ALL THE GREEN CHECKS!

**Comment by advayDev1** [comment]
also the only meaningful configuration locked behind my API user is a) whether to run on PRs or commits (I turned everything on).  everything else is configured with this file in source.

**Comment by brunobowden** [comment]
This is awesome Advay. Great stuff.

**Pull request merged**